web/source/build.sh

```diff
@@ -0,0 +1,233 @@
+
+: ${CLOSURECOMPILERPATH:=../tools}
+: ${JAVA:=java}
+
+compiler=$CLOSURECOMPILERPATH/compiler.jar
+compilecmd="$JAVA -jar $compiler"
+
+if ! [ -f $compiler ];
+then
+    echo File $compiler does not exist:  have you set the environment variable \$CLOSURECOMPILERPATH?
+fi
+
+readonly compiler
+readonly compilecmd
+
+# Establish default build parameters
+
+BUILD_UI=true
+BUILD_EMBED=true
+BUILD_FULLWEB=true
+BUILD_DEBUG_EMBED=false
+BUILD_COREWEB=true
+
+if [[ $# = 0 ]]; then
+    FULL_BUILD=true
+else
+    FULL_BUILD=false
+fi
+
+# Parse args
+if [[ $# -gt 0 ]] ; then
+    key="$1"
+    case $key in
+        -ui)
+            BUILD_EMBED=false
+            BUILD_FULLWEB=false
+            BUILD_COREWEB=false
+            ;;
+        -test)
+            BUILD_TEST=true
+            BUILD_UI=false
+            BUILD_EMBED=false
+            BUILD_FULLWEB=false
+            ;;
+        -embed)
+            BUILD_FULLWEB=false
+            BUILD_UI=false
+            BUILD_COREWEB=false
+            ;;
+        -web)
+            BUILD_EMBED=false
+            ;;
+        -debug_embedded)
+            BUILD_EMBED=false
+            BUILD_UI=false
+            BUILD_COREWEB=false
+            BUILD_FULLWEB=false
+            BUILD_DEBUG_EMBED=true
+            ;;
+        -h|-?)
+            display_usage
+            ;;
+    esac
+    shift # past argument
+fi
+
+readonly BUILD_UI
+readonly BUILD_EMBED
+readonly BUILD_FULLWEB
+readonly BUILD_DEBUG_EMBED
+readonly BUILD_COREWEB
+
+if [ $FULL_BUILD = true ]; then
+    echo Compiling build $BUILD
+    echo ""
+fi
+
+
+if [ $BUILD_EMBED = true ]; then
+    echo Compile KMEI/KMEA build $BUILD
+
+    # Create our entire embedded compilation results path.
+    if ! [ -d $EMBED_OUTPUT/resources ]; then
+        mkdir -p "$EMBED_OUTPUT/resources"  # Includes base folder, is recursive.
+    fi
+
+    # Compile supplementary plane string handing extensions
+    echo Compile SMP string extensions
+    rm $EMBED_OUTPUT/kmw-smpstring.js 2>/dev/null
+    $compilecmd --js $SOURCE/kmwstring.js --compilation_level SIMPLE_OPTIMIZATIONS --js_output_file $EMBED_OUTPUT/kmw-smpstring.js --warning_level VERBOSE
+    assert $EMBED_OUTPUT/kmw-smpstring.js
+
+    rm kmwtemp.js 2>/dev/null
+    $compilecmd --define __BUILD__=$BUILD --externs $SOURCE/kmwreleasestub.js --js $SOURCE/kmwbase.js --js $SOURCE/keymanweb.js --js $SOURCE/kmwosk.js --js $SOURCE/kmwembedded.js --js $SOURCE/kmwcallback.js --js $SOURCE/kmwkeymaps.js --js $SOURCE/kmwlayout.js --js $SOURCE/kmwinit.js --compilation_level SIMPLE_OPTIMIZATIONS  --js_output_file kmwtemp.js --warning_level VERBOSE
+    assert kmwtemp.js 
+
+    echo Append SMP extensions
+    cat $EMBED_OUTPUT/kmw-smpstring.js kmwtemp.js > $EMBED_OUTPUT/keyman.js
+    rm kmwtemp.js
+
+    echo Compiled embedded application saved as $EMBED_OUTPUT/keyman.js
+
+    # Update any changed resources
+
+    # echo Copy or update resources
+    cp -R $SOURCE/resources/ $EMBED_OUTPUT/ >/dev/null
```

Comment:
Use  `cp -Rf` to correspond to bat version of `xcopy /E /Y`

Same of other uses of cp